### ISSUE TITLE
opensuse zeek repos don't install on Fedora-40 - key error

### ISSUE THREAD

**Issue by jhaar** [zeek/zeek#3851]
Hi there

I just tried installing Zeek on Fedora-40 according to the zeek.org instructions:

dnf config-manager --add-repo https://download.opensuse.org/repositories/security:zeek/Fedora_40/security:zeek.repo
yum -y install zeek

This downloads - but fails to install due to a signing error. Looks like the gpg key in /etc/yum.repos.d/security:zeek.repo isn't the correct key? Here's a snippet of the errors

...
[SKIPPED] zeekctl-6.2.1-1.1.x86_64.rpm: Already downloaded                                                                                                                               
error: Verifying a signature using certificate AAF3EB044C49C402A9E7B9AE69D1B2AAEE3D166A (security OBS Project <security@build.opensuse.org>):
  1. Certificiate 69D1B2AAEE3D166A invalid: certificate is not alive
      because: The primary key is not live
      because: Expired on 2023-03-08T01:19:32Z
  2. Key 69D1B2AAEE3D166A invalid: key is not alive
      because: The primary key is not live
      because: Expired on 2023-03-08T01:19:32Z
...
The GPG keys listed for the "The Zeek Network Security Monitor. (Fedora_40)" repository are already installed but they are not correct for this package.
Check that the correct key URLs are configured for this repository.. Failing package is: libbroker-devel-6.2.1-1.1.x86_64
 GPG Keys are configured as: https://download.opensuse.org/repositories/security:/zeek/Fedora_40/repodata/repomd.xml.key
Public key for zeek-6.2.1-1.1.x86_64.rpm is not trusted. Failing package is: zeek-6.2.1-1.1.x86_64

**Comment by awelzel** [comment]
@jhaar - this works in a `fedora:40` container image - I'm closing this with the assumption it's something on your end.

```
# dnf install dnf-plugins-core 
# dnf config-manager --add-repo https://download.opensuse.org/repositories/security:zeek/Fedora_40/security:zeek.repo
Adding repo from: https://download.opensuse.org/repositories/security:zeek/Fedora_40/security:zeek.repo
# yum -y install zeek     
```

---

This is likely also easier to debug on Slack.

**Comment by jhaar** [comment]
Found and fixed the problem, so I'll post it here.

I was on Fedora-38 and upgraded to Fedora-40. It appears if you had the old FC38 gpg key loaded, doing "rpm --import fc40-version" didn't give an error - but didn't load the pubkey. I had to find the old one, remove it and then "rpm --import" worked correctly. i.e.

`rpm -qa gpg-pubkey --qf 'gpg-pubkey-%{version}-%{release} %{summary}\n'
gpg-pubkey-9867c58f-601c49ca Fedora (35) <fedora-35-primary@fedoraproject.org> public key
gpg-pubkey-ee3d166a-5fe932a4 security OBS Project <security@build.opensuse.org> public key
gpg-pubkey-b88c500b-5aebe12b Crowdstrike, Inc (falcon-sensor installer key) <support@crowdstrike.com> public key
gpg-pubkey-38ab71f4-60242b08 Fedora (36) <fedora-36-primary@fedoraproject.org> public key
gpg-pubkey-5323552a-6112bcdc Fedora (37) <fedora-37-primary@fedoraproject.org> public key
gpg-pubkey-18b8e74c-62f2920f Fedora (39) <fedora-39-primary@fedoraproject.org> public key
gpg-pubkey-a15b79cc-63d04c2c Fedora (40) <fedora-40-primary@fedoraproject.org> public key`

`rpm -e gpg-pubkey-ee3d166a-5fe932a4`

`rpm --import https://download.opensuse.org/repositories/security:/zeek/Fedora_40/repodata/repomd.xml.key`

...then "yum install zeek" worked fine.
